jax/_src/numpy/indexing.py

```diff
@@ -397,7 +397,8 @@ def replace(tup, val):
 
 
 def _make_along_axis_idx(shape, indices, axis):
-  return tuple_replace(lax_numpy.indices(shape, sparse=True), axis, indices)
+  axis = canonicalize_axis(axis, len(shape))
+  return tuple_update(lax_numpy.indices(shape, sparse=True), axis, indices)
```

Comment:
PRs should address a single thing: this cleanup in this file is unrelated to the main change in this PR, and should be reverted.